config/alerts.yaml

```diff
@@ -88,3 +88,121 @@
+  description: Etcd memory usage increased significantly
+  severity: warning
+
+- expr: (sum(deriv(process_resident_memory_bytes{job="etcd"}[5m]))) * 100 / sum(node_memory_MemTotal_bytes) > 10
+  description: Etcd memory usage increased significantly
+  severity: warning
+
+# API servers CPU and memory usage increase
+
+# Openshift API server
+
+#- expr: increase(container_cpu_usage_seconds_total{namespace='openshift-apiserver', container='openshift-apiserver'}[5m]) > 10
+#  description: openshift apiserver cpu usage increased
+#  severity: warning
+
+- expr: sum(rate(container_cpu_usage_seconds_total{image!='', namespace='openshift-apiserver', container='openshift-apiserver'}[1m])) * 100 / sum(machine_cpu_cores) > 10
+  description: openshift apiserver cpu usage increased significantly
+  severity: warning
+
+#- expr: deriv(container_memory_usage_bytes{namespace='openshift-apiserver', container='openshift-apiserver'}[5m]) > 10
+#  description: openshift apiserver memory usage increased
+#  severity: warning
+
+- expr: (sum(deriv(container_memory_usage_bytes{namespace='openshift-apiserver', container='openshift-apiserver'}[5m]))) * 100 / sum(node_memory_MemTotal_bytes) > 10
+  description: openshift apiserver memory usage increased significantly
+  severity: warning
+
+# Openshift kube API server
+
+#- expr: increase(container_cpu_usage_seconds_total{namespace='openshift-kube-apiserver', container='kube-apiserver'}[5m]) > 10
+#  description: openshift apiserver cpu usage increased
+#  severity: warning
+
+- expr: sum(rate(container_cpu_usage_seconds_total{image!='', namespace='openshift-kube-apiserver', container='kube-apiserver'}[1m])) * 100 / sum(machine_cpu_cores) > 10
+  description: openshift apiserver cpu usage increased significantly
+  severity: warning
+
+#- expr: deriv(container_memory_usage_bytes{namespace='openshift-kube-apiserver', container='kube-apiserver'}[5m]) > 10
+#  description: openshift apiserver memory usage increased
+#  severity: warning
+
+- expr: (sum(deriv(container_memory_usage_bytes{namespace='openshift-kube-apiserver', container='kube-apiserver'}[5m]))) * 100 / sum(node_memory_MemTotal_bytes) > 10
+  description: openshift apiserver memory usage increased significantly
+  severity: warning
+
+# Master node CPU and memory usage increase
+
+#- expr: sum(increase(container_cpu_usage_seconds_total{container!=""}[5m])) by (node) * on(node) group_left() kube_node_role{role="master"} > 10
+#  description: openshift apiserver cpu usage increased
+#  severity: warning
+
+# Master node CPU usage increase
+- expr: (sum((sum(deriv(pod:container_cpu_usage:sum{container="",pod!=""}[5m])) BY (namespace, pod) * on(pod, namespace) group_left(node) (node_namespace_pod:kube_pod_info:)  )  *  on(node) group_left(role) (max by (node) (kube_node_role{role="master"})))) * 100 / sum(machine_cpu_cores) > 10
```

Comment:
is double sum needed as above?

Reply:
Yes, in this case the inner sum groups the information and selects those which are master nodes and second sum sums this. But this is just for the master nodes